### PR TITLE
Footnote-misc: Reduce likelihood of ID collisions

### PR DESCRIPTION
**Pull request by mango-db**
[JIRA](https://jira.mongodb.org/browse/DOP-1622)

## Use case

Previously, if writer were to use a footnote, e.g. `[#mongodb-compatibility]_`, it would take the ID `mongodb-compatibility`. In that case, if the writer were to also use a header link with the ID `mongodb-compatibility`, the user would experience unexpected behavior when clicking links within the page. 

To reduce the chance of accidental ID collision, I have append `"footnote-"` to footnote IDs, as well as a unique identifier based on names to the footnote reference IDs. 

### Bonus

Handle hanging footnotes (i.e., unpaired footnotes that do not have 1 or more footnote references).

## Staging
[Datalake/index -- with add'l footnotes](https://docs-mongodborg-staging.corp.mongodb.com/DOP-931/datalake/cgoecknerwald/footnote-misc/)
[Datalake/reference -- with add'l footnotes](https://docs-mongodborg-staging.corp.mongodb.com/DOP-931/datalake/cgoecknerwald/footnote-misc/reference)
[Drivers/java](https://docs-mongodborg-staging.corp.mongodb.com/master/drivers/cgoecknerwald/footnote-misc/java/)
[Manual -- #slow-oplogs](https://docs-mongodbcom-staging.corp.mongodb.com/master/manual/cgoecknerwald/footnote-misc/reference/log-messages#footnote-slow-oplogs)

### CODE REVIEW

File: tests/unit/__snapshots__/Footnote.test.js.snap [hunk 1]
```diff
@@ -22,7 +22,7 @@ exports[`renders correctly 1`] = `
         [
         <a
           className="fn-backref"
-          href="#ref-undefined"
+          href="#ref-1-undefined"
```
Comment (sophstad) [inline tests/unit/__snapshots__/Footnote.test.js.snap:25]: Huh, why does `undefined` appear in this test at all?

Comment (mango-db) [reply tests/unit/__snapshots__/Footnote.test.js.snap:25]: In this test, there is no `FootnoteReference` for the `Footnote` to link to -- I've added a conditional to remove the `href` attribute for `Footnote`s without accompanying `FootnoteReference`s.

**Comment by mango-db** [comment]
## Updated Staging
[Datalake/index -- with add'l footnotes](https://docs-mongodborg-staging.corp.mongodb.com/footnotes-misc/datalake/cgoecknerwald/footnote-misc/)
[Datalake/reference -- with add'l footnotes](https://docs-mongodborg-staging.corp.mongodb.com/footnotes-misc/datalake/cgoecknerwald/footnote-misc/reference)
[Datalake/administration -- with add'l footnotes](https://docs-mongodborg-staging.corp.mongodb.com/footnotes-misc/datalake/cgoecknerwald/footnote-misc/administration)
[Drivers/java -- with updated footnote names](https://docs-mongodborg-staging.corp.mongodb.com/master/drivers/cgoecknerwald/footnote-misc/java/)
[Manual](https://docs-mongodborg-staging.corp.mongodb.com/master/manual/cgoecknerwald/footnote-misc/)

File: tests/unit/__snapshots__/Footnote.test.js.snap (hunk 1 above)
Comment (sophstad) [reply tests/unit/__snapshots__/Footnote.test.js.snap]: Ah, I see -- I think the appropriate approach here would be to wrap the `Footnote` component in a `FootnoteContext.Provider` with some mock data so we can actually replicate what the component is doing! You can see an example of that [here|https://github.com/mongodb/snooty/blob/master/tests/unit/Pills.test.js#L9-L11].

**Comment by mango-db** [comment]
Staging has been updated to reflect changes.

**Review by sophstad — APPROVED** [review]

**Pull request merged**